protocol/messenger_wallet.go

```diff
@@ -214,6 +213,42 @@ func (m *Messenger) deleteKeystoreFileForAddress(address types.Address) error {
 	return nil
 }
 
+func (m *Messenger) deleteKeystoreFilesForKeypair(keypair *accounts.Keypair) (err error) {
+	if keypair == nil {
+		return
+	}
+
+	if len(keypair.Keycards) == 0 {
```

Comment:
nit could be a gaurd:
if len(keypair.Keycards) != 0 {
  return
}

... the rest